src/Microsoft.SqlTools.ServiceLayer/Rename/Requests/ProcessRenameEditRequest.cs

```diff
@@ -0,0 +1,22 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+using Microsoft.SqlTools.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.Rename.Requests
+{
+    /// <summary>
+    /// Property class for Rename Service
+    /// </summary>
+    public class ProcessRenameEditRequestParams : GeneralRequestDetails
+    {
```

Comment:
the simplest parameter to pass to identify any SMO object is its Urn. I think you could just pass the Urn and the new name.
Use Server.GetSmoObject(urn) to get the object instance.

The only pain point is SMO doesn't appear to define an interface for "Rename" so you will need to have a case statement on the type or use reflection to get the Reflect method to call.